server/services/store/sqlstore/board_insights.go

```diff
@@ -0,0 +1,215 @@
+	/**
+	Some squirrel issues to note here are
+	1. https://github.com/Masterminds/squirrel/issues/285 - since we're using 1+ sub queries. When placeholders are counted for second query, the placeholder names are repeated.
+		This is the reason to not use conditional operators in Where clauses which would eventually parametrize the variables.
+	*/
+	insightsQueryStr := fmt.Sprintf(`select
+		id,
+		title,
+		icon,
+		sum(count) as activity_count,
+		%s as active_users,
+		created_by
+	from
+		(
+			select
+				boards.id,
+				boards.title,
+				boards.icon,
+				count(boards_history.id) as count,
+				boards_history.modified_by,
+				boards.created_by
+			from
+				%[2]vboards_history as boards_history
+				join %[2]vboards as boards on boards_history.id = boards.id
+			where
+				boards_history.insert_at > %[3]s
+				and boards.team_id = %[4]s
+				and boards_history.modified_by != 'system'
+				and boards.delete_at = 0
+			group by
+				boards_history.id,
+				boards.id,
+				boards_history.modified_by
+			UNION
+			ALL
+			select
+				boards.id,
+				boards.title,
+				boards.icon,
+				count(blocks_history.id) as count,
+				blocks_history.modified_by,
+				boards.created_by
+			from
+				%[2]vblocks_history as blocks_history
+				join %[2]vboards as boards on blocks_history.board_id = boards.id
+			where
+				blocks_history.insert_at > %[5]s
+				and boards.team_id = %[6]s
+				and blocks_history.modified_by != 'system'
+				and boards.delete_at = 0
+			group by
+				blocks_history.board_id,
+				boards.id,
+				blocks_history.modified_by
+		) as boards_and_blocks_history
+	group by
+		id,
+		title,
+		icon,
+		created_by
+	order by
+		activity_count desc
+	limit
+		4;`,
+		s.concatenationSelector("distinct modified_by", ","),
+		s.tablePrefix,
+		s.parameterPlaceholder(1),
+		s.parameterPlaceholder(2),
+		s.parameterPlaceholder(3),
+		s.parameterPlaceholder(4),
+	)
+
+	var args []interface{}
+	args = append(args, s.durationSelector(duration), teamID, s.durationSelector(duration), teamID)
+	rows, err := db.Query(insightsQueryStr, args...)
+
+	if err != nil {
+		s.logger.Error(`Team insights query ERROR`, mlog.Err(err))
+		return nil, err
+	}
+	defer s.CloseRows(rows)
+
+	boardsInsights, err := boardsInsightsFromRows(rows)
+	if err != nil {
+		return nil, err
+	}
+	return boardsInsights, nil
+}
+
+func (s *SQLStore) getUserBoardsInsights(db sq.BaseRunner, userID string, duration string) ([]*model.BoardInsight, error) {
+	/**
+	Some squirrel issues to note here are
+	1. https://github.com/Masterminds/squirrel/issues/285 - since we're using 1+ sub queries. When placeholders are counted for second query, the placeholder names are repeated.
+	2. No handlers at the moment for nested conditions with 'in' operator in squirrel - for the final where clause to shortlist user's boards.
+	*/
+	insightsQueryStr := fmt.Sprintf(`select *
+	from (select
```

Comment:
Isn't there any other way to do it right with squirrel, this queries looks like a lot of complexity, actually, I can see it more aceptable if the subqueries are generated independently and then you build the final query using that subquery already calculated SQL. This wall of formatted text is too much for me.

Reply:
Fixed this.